spec/unit/facter/network_nexthop_ip_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe Facter.fact(:network_nexthop_ip) do
+  before { Facter.clear }
+  after { Facter.clear }
+
+  describe 'network_nexthop_ip' do
+    context' returns gateway' do
+      before do
+        allow(Facter.fact(:kernel)).to receive(:value).and_return(:linux)
+        allow(Facter::Util::Resolution).to receive(:which).with('ip').and_return('/usr/bin/ip')
+      end
+      let(:facts) { { kernel: :linux } }
+      it { expect(Facter::Util::Resolution).to receive(:exec).with('ip route show 0/0').and_return('default via 192.168.178.1 dev eth0') }
+      it { expect(Facter.value(:network_nexthop_ip)).to eq('192.168.178.1') }
```

Comment:
I'd suggest to use the subject approach from https://github.com/puppetlabs/pdk-templates/blob/main/object_templates/fact_spec.erb for better readability.